### PR TITLE
[gbinder] Make stability field of local object configurable. JB#61912

### PR DESCRIPTION
**Pull request by NotKit**
AIDL HALs require stability field value to be set to VINTF as opposed to the default SYSTEM, so expose a way to let the caller set the value to used by finish_flatten_binder per local object.

### CODE REVIEW

File: include/gbinder_local_object.h [hunk 1]
```diff
@@ -61,6 +61,15 @@ GBinderLocalReply*
 gbinder_local_object_new_reply(
     GBinderLocalObject* obj);
 
+GBINDER_STABILITY_LEVEL
+gbinder_local_object_get_stability(
+    GBinderLocalObject* self);
```
Comment (monich) [inline include/gbinder_local_object.h:66]: Please add `/* Since 1.0.40 */` comment (here and in the .c file) so that the person reading this code knows which version of libgbinder introduced this API (and therefore which libgbinder version the program would require if it uses this API).

File: src/gbinder_rpc_protocol.c [hunk 2]
```diff
@@ -218,7 +218,7 @@ gbinder_rpc_protocol_aidl3_finish_flatten_binder(
     void* out,
     GBinderLocalObject* obj)
 {
-    *(guint32*)out = GBINDER_STABILITY_SYSTEM;
+    *(guint32*)out = gbinder_local_object_get_stability(obj);
```
Comment (monich) [inline src/gbinder_rpc_protocol.c:221]:
```
src/gbinder_rpc_protocol.c: In function ‘gbinder_rpc_protocol_aidl3_finish_flatten_binder’:
src/gbinder_rpc_protocol.c:221:22: warning: implicit declaration of function ‘gbinder_local_object_get_stability’ [-Wimplicit-function-declaration]
  221 |     *(guint32*)out = gbinder_local_object_get_stability(obj);
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: include/gbinder_local_object.h [hunk 3]
```diff
@@ -61,6 +61,15 @@ GBinderLocalReply*
 gbinder_local_object_new_reply(
     GBinderLocalObject* obj);
 
+GBINDER_STABILITY_LEVEL
+gbinder_local_object_get_stability(
+    GBinderLocalObject* self); /* Since 1.0.40 */
```
Comment (monich) [inline include/gbinder_local_object.h:66]: Is there a good  reason (or any reason at all) to make this function external? If not, then there's no need to have this function at all, even as an internal one - since  `gbinder_rpc_protocol.c` needs to include `gbinder_local_object_p.h` anyway and then it can just directly access the `stability` field (checking the object pointer for `NULL` first). I don't like exposing unnecessary API because unnecessary things have a tendency to pile up and become a burden from the backward compatibility POV.

In other words, I suggest to get rid of it))

Comment (NotKit) [reply include/gbinder_local_object.h:66]: You're right, I don't think it's needed in the external code. Just wanted it to be symmetric with set/get, but can get rid of it then.

**Review by mlehtima — APPROVED** [review]

**Review by monich — APPROVED** [review]
LGTM
